.github/workflows/publish.yml

```
@@ -6,6 +6,16 @@ on:
     tags: ['*']
 
 jobs:
+  version-check:
+    # We need this job to run only on push with tag.
+    if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
```

There is no other event types:
```suggestion
    if: startsWith(github.ref, 'refs/tags/')
```